classes/components/forms/submission/ChangeSubmissionLanguageMetadataForm.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * @file classes/components/form/submission/ChangeSubmissionLanguageMetadataForm.php
+ *
+ * Copyright (c) 2023 Simon Fraser University
+ * Copyright (c) 2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ChangeSubmissionLanguageMetadataForm
+ *
+ * @brief A form for changing submission's metadata after language change.
+ */
+
+namespace PKP\components\forms\submission;
+
+use APP\facades\Repo;
+use APP\publication\Publication;
+use APP\submission\Submission;
+use PKP\components\forms\Field;
+use PKP\components\forms\FieldOptions;
+use PKP\components\forms\FieldRadioInput;
+use PKP\components\forms\FormComponent;
+use PKP\components\forms\publication\PKPMetadataForm;
+use PKP\components\forms\publication\TitleAbstractForm;
+use PKP\context\Context;
+use PKP\facades\Locale;
+
+define('FORM_CHANGE_SUBMISSION_LANGUAGE_METADATA', 'changeSubmissionLanguageMetadata');
+
+class ChangeSubmissionLanguageMetadataForm extends FormComponent
+{
+    public $id = FORM_CHANGE_SUBMISSION_LANGUAGE_METADATA;
+    public $method = 'PUT';
+
+    public function __construct(string $action, Submission $submission, Publication $publication, Context $context, array $locales, ?string $metadataVocabSuggestionUrlBase)
+    {
+        $this->action = $action;
+        $this->locales = $locales;
+
+        $this->addGroup([
+            'id' => 'lang',
+        ]);
+
+        $submissionLocale = $submission->getData('locale');
+        $supportedLocaleNames = collect($context->getSupportedSubmissionLocaleNames());
```

Comment:
Shouldn't here publication->getLanguageNames() or submission->getPublicationLanguageNames() be used?

Reply:
I'm not sure, shouldn't this be the same list as in submission wizard where you choose the submission locale ?

Reply:
Hmmm... I am not sure as well :-D
I think it is a rare case, but: Lets say an author entered a submission with a submission full text file in Finish, and filled out metadata in German and English, and has chosen wrongly English (instead of Finish) as primary locale. Then the JM removes Finish from submission languages. Then an editor wants to fix the submission language for this submission and to chose Finish. It would not be possible, right?
It is maybe slightly different than submission wizard, because the submission could exist in the system for a while (having the possibility that submission languages have changed in the mean time), before the editor realizes that the submission primary language is wrong, no?

Reply:
Lets leave it as it is for now...